flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/autoscaler/ScalingMetricCollector.java

```diff
@@ -122,7 +130,15 @@ public CollectedMetrics getMetricsHistory(
 
         // Add scaling metrics to history if they were computed successfully
         scalingMetricHistory.put(clock.instant(), scalingMetrics);
-        scalingInformation.updateMetricHistory(currentJobStartTs, scalingMetricHistory);
+        scalingInformation.updateMetricHistory(currentJobUpdateTs, scalingMetricHistory);
+
+        if (currentJobUpdateTs
+                .plus(stabilizationDuration)
+                .isAfter(clock.instant().minus(metricsWindowDuration))) {
```

Comment:
I would prefer to have a config setting for a minimum metrics window (that should default to the metrics window), so that we can have quick scaling decisions after stabilization.

Ideally users would use longer metrics windows (15-30 minutes) and a minimum window of something much shorter like 5 mins.

This is very beneficial also for testing new changes to the algorithm, or when tuning the paramaters.

Reply:
That's a great idea. I've added another commit which does that. I might still create an extra JIRA for it to document the new feature set.

Reply:
Created [FLINK-30510](https://issues.apache.org/jira/browse/FLINK-30510) and tagged the commit.

Reply:
Looks good! Please also update the autoscaler doc page section about the improved configuration (https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/custom-resource/autoscaler/#stabilization-and-metrics-collection-intervals) and the example configs in docs and yamls.

Reply:
After thinking about this change more, it might actually be tricky for users to set the minimum metric window size correctly. There could be unexpected side effects if the delta between the two is too high, e.g. oscillation where the minimum window scales up, the maximum window scales down, and so on. It seems we only want to permit the minimum window for the initial scaling but not for subsequent scaling decisions. We could keep `metrics.window` and introduce `metrics.window.initial` which is only respected as long as the last scaling decision didn't come from the regular window.

Reply:
I think the min/max is actually simpler to understand and configure . If oscillation occurs then your stabilization period may be too small?

the default for min/max could be the same duration but I would not make this more complicated

Reply:
At any point the data rate might suddenly increase (during any scale operation) the minimum window allows us to react to it. This is not only about the first time we start the job

Reply:
I think in this early state it’s better to have flexibility with the configuration and document recommendations than to restrict the behavior. We still have time to simplify this in the future as we gather working knowledge